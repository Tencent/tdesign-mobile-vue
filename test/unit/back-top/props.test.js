/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { mount } from '@vue/test-utils';
import BackTop from '../../../src/back-top';
import { h } from 'vue';

describe('BackTop', () => {
  describe(':props', () => {
    // 是否绝对定位固定到屏幕右下方: true
    it('fixed', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","fixed":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('icon', () => {
      const fn = jest.fn();
      const wrapper = mount(BackTop, {
        propsData: {
          icon: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('icon', () => {
      const fn = jest.fn();
      const wrapper = mount(BackTop, {
        propsData: {
          icon: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 定位滚动到指定对象: 
    it('target', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","target":""},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 预设的样式类型: round
    it('theme', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","theme":"round"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 预设的样式类型: half-round
    it('theme', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","theme":"half-round"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 预设的样式类型: round-dark
    it('theme', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","theme":"round-dark"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 预设的样式类型: half-round-dark
    it('theme', () => {
      const wrapper = mount(BackTop, {
        propsData: {"text":"返回","theme":"half-round-dark"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
  });
});