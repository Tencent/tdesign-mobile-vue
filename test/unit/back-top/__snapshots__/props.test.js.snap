// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop :props fixed 1`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;

exports[`BackTop :props icon 1`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round"
>
  <div>
    text
  </div>
  <!--v-if-->
</div>
`;

exports[`BackTop :props icon 2`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <!--v-if-->
</div>
`;

exports[`BackTop :props target 1`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;

exports[`BackTop :props theme 1`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;

exports[`BackTop :props theme 2`] = `
<div
  class="t-back-top t-is-fixed t-back-top--half-round"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;

exports[`BackTop :props theme 3`] = `
<div
  class="t-back-top t-is-fixed t-back-top--round-dark"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;

exports[`BackTop :props theme 4`] = `
<div
  class="t-back-top t-is-fixed t-back-top--half-round-dark"
>
  <svg
    class="t-icon t-icon-backtop"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M2 3h12V2H2v1zM3.38 10.23l4.1-4.03v8.64H8.5V6.2l4.18 4.08.71-.7-5.05-4.93a.5.5 0 00-.7 0l-4.98 4.9.72.69z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-back-top__text"
  >
    返回
  </span>
</div>
`;
